vault/expiration.go

```diff
@@ -569,6 +589,11 @@ func (m *ExpirationManager) Stop() error {
 		m.leaseCount--
 		return true
 	})
+	m.nonexpiring.Range(func(key, value interface{}) bool {
+		m.nonexpiring.Delete(key)
+		return true
+	})
```

Comment:
Do we need to itterate over the list or can we just re-assign the nonexpiring variable to a newly created sync.Map?

Reply:
Hm, I assumed the behavior was similar enough to pending that we wanted to treat it in the same way.  But nothing gets looked up in nonexpiring in the data path.

Reply:
OK, I remembered (or re-reasoned) about why it is done this way.

Replacing m.pending and m.nonexpiring with a new value is a race condition because WalkTokens doesn't grab pendingLock before accessing them. That's the race condition Jeff warned me about in the initial API design. I added a comment explaining this.

However, I do think it's possible we could grab a pointer to the sync.Map with the lock held and then use it without the lock. I didn't introduce that in this PR.